gateway/src/apicast/policy/3scale_batcher/3scale_batcher.lua

```diff
@@ -8,16 +8,27 @@ local reporter = require('reporter')
 local Transaction = require('transaction')
 local http_ng_resty = require('resty.http_ng.backend.resty')
 local semaphore = require('ngx.semaphore')
+local TimerTask = require('resty.concurrent.timer_task')
 
 local ipairs = ipairs
 
 local default_auths_ttl = 10
 local default_batch_reports_seconds = 10
 
-local _M = policy.new('Caching policy')
+local _M, mt = policy.new('Caching policy')
 
 local new = _M.new
 
+mt.__gc = function(self)
+  -- Instances of this policy are garbage-collected when the config is
+  -- reloaded. We need to ensure that the TimerTask instance schedules another
+  -- run before that so we do not leave any pending reports.
+
+  if self.timer_task then
+    self.timer_task:schedule_one(1)
+  end
```

Comment:
And we will rely on cancelling the timer by __gc callbacks ?

Reply:
That works, but let's cancel the timer_task here to be explicit about it.

Reply:
Done :+1: